CRM/Mailing/Event/BAO/Opened.php

```diff
@@ -283,7 +283,10 @@ public static function &getRows(
 
     $query .= $groupBy;
 
-    $orderBy = "sort_name ASC, {$open}.time_stamp DESC";
+    $orderBy = "sort_name ASC ";
+    if (!$is_distinct) {
+      $orderBy .= ", {$open}.time_stamp DESC";
```

Comment:
This doesn't make sense if we are asking for distinct row as then we will only have 1 timestamp for each contact

Reply:
@seamuslee001 I think this would be the one line fix:
```patch
diff --git a/CRM/Mailing/Event/BAO/Opened.php b/CRM/Mailing/Event/BAO/Opened.php
index 5e21a4e..c0123f4 100644
--- a/CRM/Mailing/Event/BAO/Opened.php
+++ b/CRM/Mailing/Event/BAO/Opened.php
@@ -250,6 +250,7 @@ class CRM_Mailing_Event_BAO_Opened extends CRM_Mailing_Event_DAO_Opened {
 
     if ($is_distinct) {
       $groupBy = " GROUP BY $queue.id ";
+      $selectClauses[] =  "$queue.id";
       $select = CRM_Contact_BAO_Query::appendAnyValueToSelect($selectClauses, "$queue.id");
     }
     else {
```
for https://test.civicrm.org/job/CiviCRM-Core-Matrix/CIVIVER=master,label=ubuntu1604/4351/testReport/junit/(root)/CRM_Mailing_BAO_QueryTest/testOpenedMailingQuery/ isn't it ?

Reply:
i don't think so the incompatiable column is `civicrm_mailing_event_opened.time_stamp` which is in the ORDER BY is not in the group By

Reply:
Oops you are right, my bad. Yup makes sense. Deleted my comment.